PIL/_imaging.pyi

```diff
@@ -0,0 +1,104 @@
+	      ) -> ImagingCore: ...  ### what are data & bbox?
+
+def crc32(buffer: Any, hilo: Tuple[int, int]=(0,0)) -> Tuple[int, int]: ...
+### buffer is bytesarray? Should Tuple[int, int] be another named type?
+
+def effect_mandelbrot(size: Size=(512,512),
+                      extent: Tuple[float, float, float, float]=(-3,-2.5,2,2.5),
+		      quality: int=100) -> ImagingCore: ...
+def effect_noise(size: Size, sigma: float=128) -> ImagingCore: ...
+def linear_gradient(mode: Mode) -> ImagingCore: ...
+def radial_gradient(mode: Mode) -> ImagingCore: ...
+def wedge(mode: Mode) -> ImagingCore: ...
+
+class ImagingCore:
+    mode = ...  # type: str
+    size = ...  # type: Size  ## close enough?
+    bands = ...  # type: int
+    id = ... ### use? size_t? int?
+    ptr = ... ###
+
+    def getpixel(self, xy: XY) -> Optional[Any]: ...
+    def putpixel(self, xy: XY, value: Any) -> None: ...
+    def pixel_access(self, readonly: int) -> Any: ...
+
+    def convert(self, mode: Mode, dither: int=..., paletteimage: Any=...) -> ImagingCore: ...  ### what is paletteimage?
+    def convert2(self, im1: ImagingCore, im2: ImagingCore) -> None: ...
+    def convert_matrix(self, mode: Mode, matrix: Union[Matrix4, Matrix12]) -> ImagingCore: ...
+    def convert_transparent(self, mode: Mode, t: Union[Tuple[int, int, int], int]) -> ImagingCore: ...  ### Name t better?
+    def copy(self) -> ImagingCore: ...
+    def crop(self, bbox: LURD) -> ImagingCore: ...
+    def expand(self, xmargin: int, ymargin: int, mode_index: int=0) -> Any: ...  ### Include default?
+    def filter(self, size: Size, divisor: float, offset: float, kernel: Any) -> ImagingCore: ...  ### what is kernel?
+    def histogram(self, tuple2: Tuple[float, float]=..., mask: ImagingCore=...) -> List[int]: ...  ### Do these have default values? Do they take None? What is tuple2?
+
+    def modefilter(self, i: int) -> Any: ...
+
+#    def offset(self): ... ### Function removed?
+
+    def paste(self, core: ImagingCore, box: LURD, coremask: ImagingCore) -> None: ...
+    def point(self, lut: Any, mode: Optional[Mode]) -> ImagingCore: ...  ### Clarify Any
+    def point_transform(self, scale: float=1.0, offset: float=0.0) -> ImagingCore: ...  ### Include defaults or not?
+    def putdata(self, data: Any, scale: float=1.0, offset: float=0.0) -> None: ...  ### Include defaults or not?
+
+    def quantize(self, colors: int=..., method: int=..., kmeans: int=...) -> ImagingCore: ...
+
+    def rankfilter(self, size: int, rank: int) -> ImagingCore: ...
+
+    def resize(self, size: Size, resample: int=...) -> ImagingCore: ...
+    def transpose(self, method: int) -> ImagingCore: ...
+    def transform2(self, box: LURD, core: ImagingCore, method: int, data: Any, resample: int=..., fill: int=...) -> None: ...  ### What is data?
+
+    def isblock(self) -> int: ...
+
+    def getbbox(self) -> Optional[LURD]: ...
+    def getcolors(self, maxcolors: int) -> List[Tuple[int, Any]]: ...  ### Any is a 'Color'?
```

Comment:
Any is Union[Int, Float, Tuple[int,int], Tuple[int,int,int],Tuple[int,int,int,int]]